src/sqlfluff/core/parser/segments/base.py

```diff
@@ -281,52 +275,79 @@ def expand(segments, parse_context):
         check_still_complete(segments, segs, ())
         return segs
 
-    @staticmethod
-    def _realign_segments(segments, starting_pos=None, meta_only=False):
-        """Realign the positions in the provided segments."""
-        seg_buffer = []
-        todo_buffer = list(segments)
-        if not todo_buffer:
-            return ()
-        # If starting pos not provided, take it from the first of the buffer.
-        running_pos = starting_pos
-        if not running_pos:
-            for seg in todo_buffer:
-                if not seg.is_meta:
-                    running_pos = seg.pos_marker
+    @classmethod
+    def _position_segments(cls, segments, parent_pos=None):
```

Comment:
This is nice code! I expected it to be really intricate and hard to follow by its nature, but the naming, comments, etc. are all 💯.